ibis/expr/api.py

```diff
@@ -379,20 +391,69 @@ def memtable(
 
     >>> t = ibis.memtable([(1, "foo"), (2, "baz")])
     >>> t
-    PandasInMemoryTable
+    InMemoryTable
       data:
         DataFrameProxy:
              col0 col1
           0     1  foo
           1     2  baz
     """
-    import pandas as pd
-
     if columns is not None and schema is not None:
         raise NotImplementedError(
             "passing `columns` and schema` is ambiguous; "
             "pass one or the other but not both"
         )
+    return _memtable_from_data(data, name=name, schema=schema, columns=columns)
+
+
+try:
+    import pyarrow as pa  # noqa: F811
```

Comment:
Could we use `lazy_singledispatch` for `_memtable_from_data` instead to make the import optional?

Reply:
Done.